ScoutSuite/providers/aws/facade/ec2.py

```diff
@@ -32,11 +32,22 @@ def __init__(self, session: boto3.session.Session, owner_id: str):
             if 'Value' not in user_data_response['UserData'].keys():
                 return None
             else:
-                value = base64.b64decode(user_data_response['UserData']['Value'])
-                if value[0:2] == b'\x1f\x8b':  # GZIP magic number
-                    return zlib.decompress(value, zlib.MAX_WBITS | 32).decode('utf-8')
-                else:
-                    return value.decode('utf-8')
+                try:
+                    value = base64.b64decode(user_data_response['UserData']['Value'])
+                    if value[0:2] == b'\x1f\x8b':  # GZIP magic number
+                        return zlib.decompress(value, zlib.MAX_WBITS | 32).decode('utf-8')
+                    else:
+                        return value.decode('utf-8')
+                except base64.binascii.Error as e:
+                    print_exception(
+                        'Encoded UserData in Base64 has an incorrect padding')
+                finally:
+                    value = base64.b64decode(user_data_response['UserData']['Value'] + "===")  # Adds extra padding at
+                    # the end to the base64 value to avoid decoding errors
+                    if value[0:2] == b'\x1f\x8b':  # GZIP magic number
```

Comment:
Here we're doubling the code. Wouldn't it be better to first validate padding, add it if missing (as described in https://gist.github.com/perrygeo/ee7c65bb1541ff6ac770), and then process the user data once that's done?

Reply:
In my first commit I just added extra padding `===` to the end of every User Data value. This way, the decode is always successful as the extra padding is removed by the decode function if it's not needed.
That would also be my preferred solution however I was not able to add a warning or a disclaimer in case the User Data string was incorrect. I would suggest to add a little disclaimer stating that the decoded data was obtained by adding extra padding at the end of the Base64 encoded string and, therefore, the resulting output could be wrong.

How do you find this solution?